core-services/src/main/java/org/zalando/nakadi/service/RepartitioningService.java

```diff
@@ -89,18 +99,44 @@ public void repartition(final EventType eventType, final int partitions)
         }
 
         LOG.info("Start repartitioning for {} to {} partitions", eventType.getName(), partitions);
-        timelineService.updateTimeLineForRepartition(eventType, partitions);
 
-        updateSubscriptionsForRepartitioning(eventType, partitions);
-
-        // it is clear that the operation has to be done under the lock with other related work for changing event type,
-        // but it is skipped, because it is quite rare operation to change event type and repartition at the same time
+        Closeable closeable = null;
         try {
-            defaultStatistic.setReadParallelism(partitions);
-            defaultStatistic.setWriteParallelism(partitions);
-            eventTypeRepository.update(eventType);
-        } catch (Exception e) {
-            throw new NakadiBaseException(e.getMessage(), e);
+            closeable = timelineSync.workWithEventType(eventType.getName(), nakadiSettings.getTimelineWaitTimeoutMs());
+            timelineService.updateTimeLineForRepartition(eventType, partitions);
+
+            updateSubscriptionsForRepartitioning(eventType, partitions);
+
+            // it is clear that the operation has to be done under the lock with other related work for changing event
+            // type, but it is skipped, because it is quite rare operation to change event type and repartition at the
+            // same time
+            try {
+                defaultStatistic.setReadParallelism(partitions);
+                defaultStatistic.setWriteParallelism(partitions);
+                eventTypeRepository.update(eventType);
+            } catch (Exception e) {
+                throw new NakadiBaseException(e.getMessage(), e);
+            }
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            LOG.error("Failed to wait for timeline switch", e);
+            throw new EventTypeUnavailableException("Event type " + eventType.getName()
+                    + " is currently in maintenance, please repeat request");
+        } catch (final TimeoutException e) {
+            LOG.error("Failed to wait for timeline switch", e);
+            throw new EventTypeUnavailableException("Event type " + eventType.getName()
+                    + " is currently in maintenance, please repeat request");
+        } catch (final InternalNakadiException | ServiceTemporarilyUnavailableException e) {
+            LOG.error("Error deleting event type " + eventType.getName(), e);
```

Comment:
do we need to process this execeptions here? if so could you adjust the message please?

Reply:
fixed 7b017fe0a